packages/manager/modules/pci/src/projects/project/storages/databases/database/advanced-configuration/advanced-configuration.controller.js

```diff
@@ -0,0 +1,150 @@
+import { FIELD_TYPES } from './advanced-configuration.constants';
+
+export default class {
+  /* @ngInject */
+  constructor($translate, CucCloudMessage, DatabaseService, $timeout) {
+    this.$translate = $translate;
+    this.CucCloudMessage = CucCloudMessage;
+    this.DatabaseService = DatabaseService;
+    this.FIELD_TYPES = FIELD_TYPES;
+    this.$timeout = $timeout;
+  }
+
+  $onInit() {
+    this.trackDashboard('advanced-configuration', 'page');
+    this.messageContainer =
+      'pci.projects.project.storages.databases.dashboard.advanced-configuration';
+    this.loadMessages();
+    this.model = this.initModelFromAdvancedConfig();
+    this.addEmptyEntry();
+  }
+
+  initModelFromAdvancedConfig() {
+    const model = [];
+    // add existing advanced configuration to model
+    Object.keys(this.advancedConfiguration).forEach((field) => {
+      const keyConfig = this.advancedConfigurationList.find(
+        (prop) => prop.name === field,
+      );
+      let value = this.advancedConfiguration[field];
+      switch (keyConfig.type) {
+        case FIELD_TYPES.boolean:
+          value = value === 'true';
+          break;
+        case FIELD_TYPES.long:
+          value = parseInt(value, 10);
+          break;
+        case FIELD_TYPES.double:
+          value = parseFloat(value);
+          break;
+        default:
+          break;
+      }
+      model.push({
+        key: keyConfig,
+        value,
+        added: true,
+        isFixed: true,
+      });
+    });
+    return model;
+  }
+
+  addEmptyEntry() {
+    this.getAddableProperties();
+    if (this.addableProperties.length > 0) {
+      this.model.push({});
+    }
+  }
+
+  getAdvancedConfigModel() {
+    const advancedJson = {};
+    this.model.forEach((entry) => {
+      if (
+        entry?.key?.name &&
+        entry?.value !== undefined &&
+        entry?.value !== null
+      )
+        advancedJson[entry.key.name] = `${entry.value}`;
+    });
+    return advancedJson;
+  }
+
+  loadMessages() {
+    this.CucCloudMessage.unSubscribe(this.messageContainer);
+    this.messageHandler = this.CucCloudMessage.subscribe(
+      this.messageContainer,
+      {
+        onMessage: () => this.refreshMessages(),
+      },
+    );
+  }
+
+  refreshMessages() {
+    this.messages = this.messageHandler.getMessages();
+  }
+
+  getAddableProperties() {
+    this.addableProperties = this.advancedConfigurationList.filter(
+      (field) => !this.model.find((config) => config?.key?.name === field.name),
+    );
+  }
+
+  onAddConfig($index) {
+    this.model[$index].added = true;
+    this.model.push({});
+    this.getAddableProperties();
+  }
+
+  onRemoveConfig($index) {
+    this.model.splice($index, 1);
+    this.getAddableProperties();
+  }
+
+  onTypeChanged(modelValue, $index) {
+    this.model[$index].key = undefined;
```

Comment:
**suggestion**: Good practice to use `null` instead of `undefined`.

Reply:
Changed: https://github.com/ovh/manager/pull/6568/commits/d41de0c46bc372d3834ffe66082fcb7eca3aa01d